<h1>Data Analytics </h1>
<hr>
<h3>Name: Harry Amante</h3>

In [1]:
import numpy as  np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Import Dataset</p>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

penguins


<hr>
<h3>Descriptive Analytics</h3>

In [3]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [4]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

In [5]:
penguins.value_counts()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  sex     year
Adelie   Biscoe  34.5            18.1           187.0              2900.0       female  2008    1
                 35.0            17.9           190.0              3450.0       female  2008    1
                                                192.0              3725.0       female  2009    1
                 35.3            18.9           187.0              3800.0       female  2007    1
                 35.5            16.2           195.0              3350.0       female  2008    1
                                                                                               ..
Gentoo   Biscoe  53.4            15.8           219.0              5500.0       male    2009    1
                 54.3            15.7           231.0              5650.0       male    2008    1
                 55.1            16.0           230.0              5850.0       male    2009    1
                 55.9    

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<hr>
<h2>Selecting Columns</h2>

In [7]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [8]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [9]:
adelie = penguins[penguins['species'] == "Adelie"]

In [10]:
adelie

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [11]:
gentoo = penguins[penguins['species'] == "Gentoo"]

In [12]:
gentoo


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [13]:
chinstrap = penguins[penguins['species'] == "Chinstrap"]

In [14]:
chinstrap

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [15]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [16]:
(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [17]:
penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [18]:
chinsttrap_heavy = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000) ]

In [19]:
chinsttrap_heavy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [20]:
gender = [penguins['sex'] == "male"]

In [21]:
gender

[0       True
 1      False
 2      False
 3      False
 4      False
        ...  
 339     True
 340    False
 341     True
 342     True
 343    False
 Name: sex, Length: 344, dtype: bool]

<hr>
<h2>How to Fill up Null Values</h2>

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [24]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [25]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [26]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [27]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [28]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [29]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [30]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

<hr>
<h2>How to Change data Types of Columns</h2>

In [31]:
penguins['sex'] = penguins['sex'].astype('category')
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')


In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<hr>
<h2>How to create Columns</h2>

In [33]:
penguins['mass_kg'] = penguins['body_mass_g'] / 1000

In [34]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [35]:
penguins


species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

In [36]:
penguins['mass_kg'] = penguins['mass_kg'].fillna(penguins['mass_kg'].mean())

<hr>
<h1>Explanatory Data Analytics(EDA)</h1>

<h2>1. Which  island has more penguins?</h2>




In [37]:
penguins['species'].unique()

['Adelie', 'Gentoo', 'Chinstrap']
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [38]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3>Insight#1</h3>
<p>Biscoe has more peguins population</p>




<hr>
<h2>2. Which sex of penguins has the highest count?</h2>

In [39]:
penguins['sex'].mode()

0    male
Name: sex, dtype: category
Categories (2, object): ['female', 'male']

In [40]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h3>Insight#2</h3>
<p>Males outnumber females based on the results with a counts of 179 male and 165 female penguin.</p>

<hr>
<h2>3. What is the average bill lenght for adelie on Togersen Island?</h2>

In [41]:
aver_bill_to = penguins[penguins['species'] == 'Adelie'][['species', 'island', 'bill_length_mm']]

In [42]:
aver_bill_to

species     island  bill_length_mm
0    Adelie  Torgersen        39.10000
1    Adelie  Torgersen        39.50000
2    Adelie  Torgersen        40.30000
3    Adelie  Torgersen        43.92193
4    Adelie  Torgersen        36.70000
..      ...        ...             ...
147  Adelie      Dream        36.60000
148  Adelie      Dream        36.00000
149  Adelie      Dream        37.80000
150  Adelie      Dream        36.00000
151  Adelie      Dream        41.50000

[152 rows x 3 columns]

In [43]:
aver_bill_to['bill_length_mm'].mean()

np.float64(38.82514427516159)

In [44]:
round(aver_bill_to['bill_length_mm'].mean(),1)

np.float64(38.8)

<h3>Insight#3</h3>
<p>The average bill lenght for adelie on Togersen Island is 38.8mm </p>

<hr>
<h2>4. How many penguins of each species are there?</h2>

In [45]:
species_counts = penguins['species'].value_counts()

In [46]:
species_counts

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>Insight#4</h3>
<p>The results show that Adelie has the highest number at 152, followed by Gentoo with 124, and Chinstrap with 68.</p>

<hr>
<h2>5. How many penguins were measured on each island in 2008?</h2>

In [47]:
island_counts_2008 = penguins[penguins['year'] == 2008]['island']

In [48]:
island_counts_2008 

50     Biscoe
51     Biscoe
52     Biscoe
53     Biscoe
54     Biscoe
        ...  
315     Dream
316     Dream
317     Dream
318     Dream
319     Dream
Name: island, Length: 114, dtype: category
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [49]:
island_counts_2008.value_counts()

island
Biscoe       64
Dream        34
Torgersen    16
Name: count, dtype: int64

<h3>Inight#5</h3>
<p>The results show that Biscoe has the highest number measured in 2008 with 64 counts, followed by Dream with 34, and Torgersen with 16</p>

<hr>
<h2>6. What is the most common bill depth among all penguins?</h2>

In [50]:
most_cmmn_bill_depth = penguins['bill_depth_mm'].mode()

In [51]:
most_cmmn_bill_depth

0    17.0
Name: bill_depth_mm, dtype: float64

In [52]:
most_cmmn_bill_depth = penguins['bill_depth_mm'].mode()[0]

In [53]:
most_cmmn_bill_depth


np.float64(17.0)

<h3>Insight#6</h3>
<p>Base on the results the most common bill depth among all penguins is 17.0 mm</p>

<hr>
<h2>7. What is the median body mass of female Gentoo penguins?
</h2>

In [54]:
body_mass_fgentoo = penguins[(penguins['species'] == 'Gentoo') & (penguins['sex'] == 'female')]['body_mass_g'].median()

In [55]:
body_mass_fgentoo

np.float64(4700.0)

<h3>Inight#7</h3>
<p>The median body mass of female Gentoo penguins is 4700.0 g</p>

<hr>
<p><strong>'groupby()'</strong> 
</p>
<h2>8. How many males and females penguins are there per islands?</h2>

In [56]:
male_femle_per_isl = penguins.groupby(['island','sex']).size()


In [57]:
male_femle_per_isl

island     sex   
Biscoe     female    80
           male      88
Dream      female    61
           male      63
Torgersen  female    24
           male      28
dtype: int64

<h3>Insight#8</h3>
<p>Base in the results there are 80 female and 88 male penguins in Bicoe, 61 female and 63 male penguins in Dream, and 24 female and 28 male penguins in Torgeren</p>


<hr>
<h2>9. Which island has penguins with shorter bill lenght/What species is it?</h2>

In [58]:
shrt_bill_lnght = penguins.groupby(['island','species'])['bill_length_mm'].min()

In [59]:
shrt_bill_lnght

island     species  
Biscoe     Adelie       34.5
           Chinstrap     NaN
           Gentoo       40.9
Dream      Adelie       32.1
           Chinstrap    40.9
           Gentoo        NaN
Torgersen  Adelie       33.5
           Chinstrap     NaN
           Gentoo        NaN
Name: bill_length_mm, dtype: float64

<h3>Insight#9</h3>
<p>Base on the results the penguins with shorter bill lenghts 32.1 mm is Adelie and reside in Dream island.</p>

<hr>
<h2>10. In which year the heviest penguin in kg recorded?</h2>

In [109]:
heviest_penguin = penguins['mass_kg'].max()
year_heviest_penguin = penguins[penguins['mass_kg'] == heviest_penguin]['year'].values[0]

heviest_penguin

In [110]:
print(f"heviest penguins in kg:{heviest_penguin}\n year:{year_heviest_penguin}")

heviest penguins in kg:6.3
 year:2007


<h3>Insight#10</h3>
<p>Base on the results the heviest penguins 6.3kg recorded in 2007 </p>

<hr>
<p><strong>'correlation'</strong> <li>a >= 0.5 = true </li>
</p>
<h2>11. Is bill depth influenced by the sex of the penguins?</h2>

In [122]:
penguins['sex_num'] = penguins['sex'].replace({'male': 1, 'female': 0})

In [126]:
cor_billdep_sex = penguins['sex_num'].corr(penguins['bill_depth_mm'])

In [127]:
round(cor_billdep_sex,1)

np.float64(0.4)

<h3>Insight#11</h3>
<p>Base on the reults 0.4 it seems the correlation between the two is weak.</p>

<hr>
<h2>12. Is there a correlation between penguins body mass and bill lenght?</h2>

In [128]:
corr_bdymass_billnght = penguins['body_mass_g'].corr(penguins['bill_length_mm'])

In [130]:
round(corr_bdymass_billnght,1)

np.float64(0.6)

<h3>Insight#12</h3>
<p>Base on the reults 0.6 it seems that there a strong correlation between the two.</p>